fakebook/src/app/component/profile-view/profile-view.component.ts

```diff
@@ -10,27 +14,75 @@ import { ProfileService } from '../../service/profile.service';
   styleUrls: ['./profile-view.component.css']
 })
 export class ProfileViewComponent implements OnInit {
-  user: User | null = null;
+  user: User | undefined;
+  selfUser!: User;
+  // Person logged in email
+  currentUserEmail = '';
+  // This should be the email of the person whose profile you click on
+  profileUserEmail = '';
+  posts: Post[] = [];
+  followStatus = false;
+  selfProfileCheck = false;
 
-  constructor(
-    private route: ActivatedRoute, // getting the id # in route
-    private profileService: ProfileService
-  ) {}
+  constructor(private oktaAuth: OktaAuthService,
+              private profileService: ProfileService,
+              private route: ActivatedRoute,
+              private followService: FollowService,
+              private postService: PostService) { }
+
+  async ngOnInit(): Promise<void> {
+    // Get Current User Email
+    const userClaims = await this.oktaAuth.getUser();
+    if (userClaims) {
+      this.currentUserEmail = userClaims.email ?? '';
+    }
 
-  ngOnInit(): void {
     this.getUser();
   }
-  getUser(): void {
-    let tempId = ''; // the only way i could declare a variable that may accept a null value in thefuture
-    // get the id number from the route
-    if (this.route.snapshot.paramMap.get('id') != null) {
-      tempId += this.route.snapshot.paramMap.get('id');
 
-      const id = tempId;
+  async getUser(): Promise<void> {
+    let tempEmail = '';
+    // Async problems possibly
+    this.profileService.GetProfile(this.currentUserEmail).subscribe(profile => this.selfUser = profile);
+
+    if (this.route.snapshot.paramMap.get('email') != null) {
+      tempEmail += this.route.snapshot.paramMap.get('email');
+
+      const email = tempEmail;
+
+      this.profileService.GetProfile(email).subscribe(user => this.user = user);
+
+      this.postService.getUserPosts(email).subscribe(posts => this.posts = posts);
+      // This might have problems due to async. currentUserEmail may not be set in time
```

Comment:
are these comments still relevant? out of date comments are very bad. just looking at this method i don't see obvious async problems. but let's discuss this if you're not certain here are no async problems.

Reply:
One of them was out of date and it is deleted, the other one related to another one of your comments where the line of code with is was not necessary and has since been deleted.